### PR TITLE
docs: multi-select combo box component

### PR DESCRIPTION
**Pull request by sissbruecker**
Adds documentation and examples for the multi-select combo box component.

~~Marked as draft, as this requires a release for 23.2. For now the PR uses `23.2-SNAPSHOT` for review / testing purposes. As soon as the latest docs version has been bumped to 23.2, the changes to `pom.xml` and `package.json` should be reverted, and the PR can be merged.~~

PR has been rebased on latest which now uses a version that includes the component.

### CODE REVIEW

File: articles/components/multi-select-combo-box/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+--
+
+The overlay opens when the user clicks the field using a pointing device.
+Using the Up/Down arrow keys or typing a character (found in at least one of the options) when the field is focused also opens the popup.
+
+
+== Common Combo Box Features
+
+Multi Select Combo Box supports the following features from the regular Combo Box. All the linked examples and code snippets can be applied by replacing the Combo Box with a Multi Select Combo Box.
```
Comment (web-padawan) [inline articles/components/multi-select-combo-box/index.asciidoc:47]: Let's use `Multi-Select` with dash for consistency.

Comment (sissbruecker) [reply articles/components/multi-select-combo-box/index.asciidoc:47]: Done

File: articles/components/multi-select-combo-box/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,230 @@
+---
+title: Multi-Select Combo Box
+page-links:
+  - 'API: https://cdn.vaadin.com/vaadin-web-components/{moduleNpmVersion:multi-select-combo-box}/#/elements/vaadin-multi-select-combo-box[TypeScript] / https://vaadin.com/api/platform/{moduleMavenVersion:com.vaadin:vaadin}/com/vaadin/flow/component/combobox/MultiSelectComboBox.html[Java]'
+  - 'Source: https://github.com/vaadin/web-components/tree/v{moduleNpmVersion:multi-select-combo-box}/packages/vaadin-multi-select-combo-box[TypeScript] / https://github.com/vaadin/flow-components/tree/{moduleMavenVersion:com.vaadin:vaadin}/vaadin-combo-box-flow-parent[Java]'
+---
+
+= Multi-Select Combo Box
```
Comment (jouni) [inline articles/components/multi-select-combo-box/index.asciidoc:8]: This avoids the Vale errors for `kbd`. This `:experimental:` attribute is set automatically for all pages during the DSP build, but not in GitHub.



```suggestion
= Multi-Select Combo Box
:experimental:
```

File: articles/components/multi-select-combo-box/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+----
+include::{root}/src/main/java/com/vaadin/demo/component/multiselectcombobox/MultiSelectComboBoxSelectionChange.java[render,tags=snippet,indent=0,group=Java]
+----
+
+[source,java]
+----
+include::{root}/src/main/java/com/vaadin/demo/domain/Country.java[group=Java,tags=*,indent=0]
+----
+
+--
```
Comment (jouni) [inline articles/components/multi-select-combo-box/index.asciidoc:118]: By moving the code example description inside the example block, we can avoid unnecessary “noise” for readers, and only show the relevant information based on the example group they’ve chosen.

```suggestion

[.example]
--

Use the `selected-items-changed` event to react to the user changing the selection:

[source,typescript]
----
include::{root}/frontend/demo/component/multi-select-combo-box/multi-select-combo-box-selection-change.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source,typescript]
----
include::{root}/frontend/generated/com/vaadin/demo/domain/Country.ts[group=TypeScript]
----

Use the `addValueChangeListener` event to react to the user changing the selection:

[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/component/multiselectcombobox/MultiSelectComboBoxSelectionChange.java[render,tags=snippet,indent=0,group=Java]
----

[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/domain/Country.java[group=Java,tags=*,indent=0]
----

--
```

File: articles/components/multi-select-combo-box/index.asciidoc [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+
+== Best Practices
+
+Multi-Select Combo Box supports lazy loading for large datasets. It reduces the initial load time, consumes less bandwidth and resources.
+
+For consistency, the default width of the Multi-Select Combo Box matches that of other Vaadin input fields.
```
Comment (jouni) [inline articles/components/multi-select-combo-box/index.asciidoc:197]: Generally, avoid using the word “Vaadin” in the context of the Design System documentation, as that makes it more generally applicable for custom design systems.

That word is obviously all over the place in the code examples, so it’s not a big issue if it is also present in the other writing. But I think we can omit it in this case without making the message less clear.

```suggestion
For consistency, the default width of the Multi-Select Combo Box matches that of other input fields.
```

File: articles/components/multi-select-combo-box/index.asciidoc (hunk 2 above)
Comment (sissbruecker) [reply articles/components/multi-select-combo-box/index.asciidoc:8]: Done

File: articles/components/multi-select-combo-box/index.asciidoc (hunk 3 above)
Comment (sissbruecker) [reply articles/components/multi-select-combo-box/index.asciidoc]: Done

File: articles/components/multi-select-combo-box/index.asciidoc (hunk 4 above)
Comment (sissbruecker) [reply articles/components/multi-select-combo-box/index.asciidoc]: Makes sense, done

File: articles/components/multi-select-combo-box/index.asciidoc [hunk 5]
```diff
@@ -0,0 +1,235 @@
+---
+title: Multi-Select Combo Box
+page-links:
+  - 'API: https://cdn.vaadin.com/vaadin-web-components/{moduleNpmVersion:multi-select-combo-box}/#/elements/vaadin-multi-select-combo-box[TypeScript] / https://vaadin.com/api/platform/{moduleMavenVersion:com.vaadin:vaadin}/com/vaadin/flow/component/combobox/MultiSelectComboBox.html[Java]'
+  - 'Source: https://github.com/vaadin/web-components/tree/v{moduleNpmVersion:multi-select-combo-box}/packages/vaadin-multi-select-combo-box[TypeScript] / https://github.com/vaadin/flow-components/tree/{moduleMavenVersion:com.vaadin:vaadin}/vaadin-combo-box-flow-parent[Java]'
```
Comment (web-padawan) [inline articles/components/multi-select-combo-box/index.asciidoc:5]:
```suggestion
  - 'Source: https://github.com/vaadin/web-components/tree/v{moduleNpmVersion:multi-select-combo-box}/packages/multi-select-combo-box[TypeScript] / https://github.com/vaadin/flow-components/tree/{moduleMavenVersion:com.vaadin:vaadin}/vaadin-combo-box-flow-parent[Java]'
```

File: articles/components/multi-select-combo-box/index.asciidoc [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+[source,java]
+----
+include::{root}/src/main/java/com/vaadin/demo/domain/Country.java[group=Java,tags=*,indent=0]
+----
+
+--
+
+The overlay opens when the user clicks the field using a pointing device.
+Using the Up/Down arrow keys or typing a character (found in at least one of the options) when the field is focused also opens the popup.
```
Comment (web-padawan) [inline articles/components/multi-select-combo-box/index.asciidoc:43]: We could use this notation (can be probably fixed later for ComboBox as well):

```suggestion
Using the kbd:[Up] and kbd:[Down]arrow keys or typing a character (found in at least one of the options) when the field is focused also opens the popup.
```

I've also seen `kbd:[↑]` and `kbd:[↓]` in grid keyboard navigation docs. Not sure which one is better.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**